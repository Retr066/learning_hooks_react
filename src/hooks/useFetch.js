import { useEffect, useRef, useState } from "react";
import { getData } from "../utils/getData";

export default function useFetch(API) {
  const initialState = {
    data: [],
    loading: true,
    error: null,
  };
  const [value, setValue] = useState(initialState);
  const isMounted = useRef(true);
  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);

  useEffect(() => {
    setValue(initialState);
    getData(API)
      .then((data) => {
        if (isMounted.current) {
          setValue({
            data,
            loading: false,
            error: null,
          });
        }
      })
      .catch(() =>
        setValue({
          ...initialState,
          error: "Algo ha pasado sulca medica",
        })
      );
  }, [API]);

  return value;
}
