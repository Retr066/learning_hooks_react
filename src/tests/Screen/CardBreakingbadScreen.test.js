import { shallow } from "enzyme";
import useCounter from "../../hooks/useCounter";
import useFetch from "../../hooks/useFetch";
import CardBreakingbad from "../../screen/CardBreakingbadScreen";
jest.mock("../../hooks/useFetch");
jest.mock("../../hooks/useCounter");
describe("Pruebas para CardBreakingbadScreen", () => {
  beforeEach(() => {
    useCounter.mockReturnValue({
      counter: 10,
    });
  });
  test("debe renderizar correctamente", () => {
    useFetch.mockReturnValue({
      data: [],
      loading: true,
      error: null,
    });
    const wrapper = shallow(<CardBreakingbad />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de renderizar la informacion correctamente", () => {
    useFetch.mockReturnValue({
      data: [
        {
          name: "jherson",
          img: "algo.png",
          status: "billonario",
          birthday: "16/08/1999",
          category: "Jefe",
          nickname: "Retr0",
          occupation: ["Programador", "Dotero", "Streamer"],
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<CardBreakingbad />);
    expect(wrapper.find(".alert").exists()).toBe(false);
    expect(wrapper.find("Card").prop("name")).toBe("jherson");
  });
});
