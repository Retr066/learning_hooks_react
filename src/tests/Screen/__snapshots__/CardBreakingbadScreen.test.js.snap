// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para CardBreakingbadScreen debe renderizar correctamente 1`] = `
<div
  className="container"
>
  <h2>
    Form con Custom Hooks
  </h2>
  <hr />
  <Alert />
</div>
`;
